### PR TITLE
ReadByUserIds functionality: new API, component modification

### PR DESCRIPTION
**Pull request by taylorfinelli**
# What type of PR is this? (Check all that apply)

- [x] ✨**feature**: Introduces completely new code or new features.
- [x] 🐛**fix**: Implements changes that fix a bug. Ideally, reference an issue if present.
- [ ] ♻️**refactor**: Includes any code-related change that is neither a fix nor a feature.
- [ ] ✅**build**: Encompasses all changes related to the build of the software, including changes to dependencies or the addition of new ones.
- [ ] ⚡️**test**: Pertains to all changes regarding tests, whether adding new tests or modifying existing ones.
- [ ] 🚰**ci**: Involves all changes related to the configuration of continuous integration, such as GitHub Actions or other CI systems.
- [ ] 📚**docs**: Includes all changes to documentation, such as README files, or any other documentation present in the repository.
- [ ] 🗑️**chore**: Captures all changes to the repository that do not fit into the above categories.

# Description

- New API
  - `markChatAsReadByChatIdAndUserIdAndChatType` 
    - Takes ChatID, UserID, and ChatType as params, pushes logged in user's ID into readByUserIds array in DB
- Modified `Conversation` component to support this API and feature
  - If the conversation is not read, onPress => push logged in user's ID into DB, push same ID into local list to immediately reflect style change
    - If we didn't push into local list, the style would not immediately change and we would need to wait for another API call
  - Reformatted with Prettier

# Tests

How was this tested?

- [ ] Unit tests
- [ ] Integration tests
- [ ] E2E tests
- [x] Manual tests
- [ ] Tests were NOT needed
- [ ] Other (explain below)

More testing when I learn how to do TRPC unit testing

# Screenshots

https://github.com/UNLV-CS472-672/2024-S-GROUP3-Barbell/assets/85365584/11b00527-15aa-4a5e-9e6b-420bbad811a1

**UserID 9 has been pushed to DB**
Screenshot 2024-04-01 at 3 10 16 PM

# Documentation

- [ ] Added to README.me
- [ ] Seperate document
- [x] NO documentation needed

### CODE REVIEW

**Review by mva919 — APPROVED** [review]
YASSS good work

**Comment by thienguen** [comment]
lgmt

**Review by thienguen — APPROVED** [review]

File: apps/expo/src/components/notif/Conversation.tsx [hunk 1]
```diff
@@ -1,47 +1,62 @@
...
+import { api } from '~/utils/api'
+
 export interface ConversationProps {
-  chatId: number,
-  messageContent: string,
-  createdAt: Date,
-  chatName: string,
-  readBy: number[],
-  type: ChatType,
+  chatId: number
```
Comment (elliotwesoff) [inline apps/expo/src/components/notif/Conversation.tsx:12]: Good practice for interfaces is to use semicolons to separate fields, for symmetry with class declarations:

https://ts.dev/style/#member-property-declarations

File: packages/api/src/router/notif.ts [hunk 2]
```diff
@@ -42,6 +42,33 @@ export const notifRouter = createTRPCRouter({
...
+  /**
+   *  @remarks
+   *  Adds current user to readByUserIds field in DB
+   *
+   *  @param  chatId - the id of the chat
+   *  @param  userId - the id of the logged in user
+   *  @param  type - the chat type (DIRECT or GROUP)
+   *  @returns an array of message objects
+   */
+  markChatAsReadByChatIdAndUserIdAndChatType: publicProcedure
```
Comment (elliotwesoff) [inline packages/api/src/router/notif.ts:54]: Is there a way to update a chat in a more generic way?

File: apps/expo/src/components/notif/Conversation.tsx (hunk 1 above)
Comment (taylorfinelli) [reply apps/expo/src/components/notif/Conversation.tsx:12]: This is how Prettier formatted it. If we want to change it then we need to change our Prettier configuration.

File: packages/api/src/router/notif.ts (hunk 2 above)
Comment (mva919) [reply packages/api/src/router/notif.ts:54]: I think its better to keep the apis seperated how they are for easier testing and implementation.

**Pull request merged**